mkt/search/filters.py

```diff
@@ -95,15 +95,20 @@ def filter_queryset(self, request, queryset, view):
 
         # Add a boost for the preferred region, if it exists.
         region = get_region_from_request(request)
-        if region:
-            should.append(query.Term(**{'preferred_regions': {
-                'value': region.id,
-                'boost': 4}}))
+        region_function = {
+            'filter': {'term': {'preferred_regions': region.id}},
+            # TODO: When we upgrade to Elasticsearch 1.4, change this to
+            # 'weight'
+            'boost_factor': 4,
+        }
```

Comment:
There's no way I could find to do this using elasticsearch-dsl so I used the raw dict.